File: Trees/Binary Tree Right Side View/recur.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void helper(TreeNode* root, vector<int> & v, int level){
        if (root==NULL) return;
        else if (level>=v.size()) v.emplace_back(root->val);
        helper(root->right,v,level+1);
        helper(root->left,v,level+1);
        return;
    }
    vector<int> rightSideView(TreeNode* root) {
        vector<int> v;
        helper(root,v, 0);
        return v;
    }
};

File: Trees&Tries/Lowest Common Ancestor of a Binary Tree/recur.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:      
    ret=None
    def lowestCommonAncestor(self, poo: 'TreeNode', p: 'TreeNode', q: 'TreeNode') -> 'TreeNode':
        start =poo
        def helper(root):
            if (root==None or self.ret!=None): return False
            one = helper(root.left)
            two = helper(root.right)
            med=False
            if (root.val==p.val or root.val==q.val): med= True
            
            if one + two+med >1:
                self.ret=root
            
            return one or two or med
            
        helper(poo)
        return self.ret